crates/bevy_ecs/src/schedule/executor_parallel.rs

```diff
@@ -188,8 +196,17 @@ impl ParallelExecutor {
                         .await
                         .unwrap_or_else(|error| unreachable!(error));
                     #[cfg(feature = "trace")]
-                    let system_span =
-                        bevy_utils::tracing::info_span!("system", name = &*system.name());
+                    let system_span = {
+                        if let Some(span) = span {
+                            bevy_utils::tracing::info_span!(
+                                parent: span,
+                                "system",
+                                name = &*system.name()
+                            )
+                        } else {
+                            bevy_utils::tracing::info_span!("system", name = &*system.name())
+                        }
+                    };
```

Comment:
As far as I can see this is an exact copy of the `SingleThreadedExecutor`'s code to calculate this - it might be worth factoring it out?